Packs/EWS/Integrations/EWSO365/EWSO365.py

```diff
@@ -0,0 +1,1757 @@
+            "credentials": credentials,
+            "auth_type": OAUTH2,
+            "version": Version(EXCHANGE_O365),
+            "service_endpoint": "https://outlook.office365.com/EWS/Exchange.asmx",
+        }
+
+        return Configuration(**config_args)
+
+    def get_account(self, target_mailbox=None, access_type=None):
+        if not target_mailbox:
+            target_mailbox = self.account_email
+        if not access_type:
+            access_type = self.access_type
+        return Account(
+            primary_smtp_address=target_mailbox,
+            autodiscover=False,
+            config=self.config,
+            access_type=access_type,
+        )
+
+    def get_items_from_mailbox(self, account, item_ids):
+        # allow user to pass target_mailbox as account
+        if isinstance(account, str):
+            account = self.get_account(account)
+        else:
+            account = self.get_account(self.account_email)
+        if type(item_ids) is not list:
+            item_ids = [item_ids]
+        items = [Item(id=x) for x in item_ids]
+        result = list(account.fetch(ids=items))
+        result = [x for x in result if not isinstance(x, ErrorItemNotFound)]
+        if len(result) != len(item_ids):
+            raise Exception(
+                "One or more items were not found. Check the input item ids"
+            )
+        return result
+
+    def get_item_from_mailbox(self, account, item_id):
+        result = self.get_items_from_mailbox(account, [item_id])
+        if len(result) == 0:
+            raise Exception(f"ItemId {str(item_id)} not found")
+        return result[0]
+
+    def get_attachments_for_item(self, item_id, account, attachment_ids=None):
+        item = self.get_item_from_mailbox(account, item_id)
+        attachments = []
+        attachment_ids = argToList(attachment_ids)
+        if item:
+            if item.attachments:
+                for attachment in item.attachments:
+                    if (
+                        attachment_ids
+                        and attachment.attachment_id.id not in attachment_ids
+                    ):
+                        continue
+                    attachments.append(attachment)
+
+        else:
+            raise Exception("Message item not found: " + item_id)
+
+        if attachment_ids and len(attachments) < len(attachment_ids):
+            raise Exception(
+                "Some attachment id did not found for message:" + str(attachment_ids)
+            )
+
+        return attachments
+
+    def is_default_folder(self, folder_path, is_public):
+        if is_public is not None:
+            return is_public
+
+        if folder_path == self.folder_name:
+            return self.is_public_folder
+
+        return False
+
+    def get_folder_by_path(self, path, account=None, is_public=False):
+        if account is None:
+            account = self.get_account()
+        # handle exchange folder id
+        if len(path) == 120:
```

Comment:
It's a magic number for the length of an exchange folder id, I'll move this to a const. Do note that if the user by chance gave a path that's 120 chars long, it'll still treat it as a path and not as an id, since it won't find the given "id" in the mailbox folder ids.